### PR TITLE
Refs Task #387354 test update user

### PR DESCRIPTION
**Pull request by nguyenthinhhn**
## Related Tickets

- [#387354](https://dev.sun-asterisk.com/issues/387354)

## What's this PR do ?

- [x] test update user

## Performance

- [ ] Resolved n + 1 query
- [ ] Run explain query already
- [ ] Time run rake task : 1000 ms

## Checklist

- [x] It was tested in local success?
- [x] Fill link PR into ticket and the opposite
- [x] Note purpose/reason, solution, scope of influence into ticket

## ENV note

```
```

## Notes
*(Other notes)*

### CODE REVIEW

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr></table>

**Review by ghost — APPROVED** [review]

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr></table>

**Review by quanghung97 — APPROVED** [review]
OK

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>failure</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr></table>

File: app/Http/Controllers/UserController.php [hunk 1]
```diff
@@ -125,4 +125,339 @@ public function destroy(User $user)
...
         }
     }
+
+    public function testUpdateUserSuccessFeature()
+    {
+        $admin = factory(User::class)->create(['role' => UserType::ADMIN]);
+        $this->actingAs($admin);
+        $user = factory(User::class)->create(
+            [
+                'name' => "Name Create",
```
Comment (framgiaciv3) [inline app/Http/Controllers/UserController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Strings.DoubleQuoteUsage.NotRequired</br>**Severity**: error</br>**File**: app/Http/Controllers/UserController.php</br>**Line**: 135</br>String "Name Create" does not require double quotes; use single quotes instead

File: app/Http/Controllers/UserController.php [hunk 2]
```diff
@@ -125,4 +125,339 @@ public function destroy(User $user)
...
+                'email' => 'email@gmail.com',
+                'password' => '12345678',
+                'role' => '1',
+                'avatar' => UploadedFile::fake()->image('avatar.jpg'),
+            ]
+        );
+        $params = [
+            'name' => "Name Update",
```
Comment (framgiaciv3) [inline app/Http/Controllers/UserController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Strings.DoubleQuoteUsage.NotRequired</br>**Severity**: error</br>**File**: app/Http/Controllers/UserController.php</br>**Line**: 143</br>String "Name Update" does not require double quotes; use single quotes instead

File: app/Http/Controllers/UserController.php [hunk 3]
```diff
@@ -125,4 +125,339 @@ public function destroy(User $user)
...
+            'messageSuccess' => 'This user successfully updated',
+        ]);
+    }
+
+    public function testUnauthenticateCannotUpdateUser()
+    {
+        $admin = factory(User::class)->create(['role' => UserType::ADMIN]);
+        $user = factory(User::class)->create(
+            [
+                'name' => "Name Create",
```
Comment (framgiaciv3) [inline app/Http/Controllers/UserController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Strings.DoubleQuoteUsage.NotRequired</br>**Severity**: error</br>**File**: app/Http/Controllers/UserController.php</br>**Line**: 160</br>String "Name Create" does not require double quotes; use single quotes instead

File: app/Http/Controllers/UserController.php [hunk 4]
```diff
@@ -125,4 +125,339 @@ public function destroy(User $user)
...
+                'email' => 'email@gmail.com',
+                'password' => '12345678',
+                'role' => '1',
+                'avatar' => UploadedFile::fake()->image('avatar.jpg'),
+            ]
+        );
+        $params = [
+            'name' => "Name Update",
```
Comment (framgiaciv3) [inline app/Http/Controllers/UserController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Strings.DoubleQuoteUsage.NotRequired</br>**Severity**: error</br>**File**: app/Http/Controllers/UserController.php</br>**Line**: 168</br>String "Name Update" does not require double quotes; use single quotes instead

File: app/Http/Controllers/UserController.php [hunk 5]
```diff
@@ -125,4 +125,339 @@ public function destroy(User $user)
...
+        $response->assertRedirect('login');
+    }
+
+    public function testUnauthorizationCannotUpdateUser()
+    {
+        $user1 = factory(User::class)->create(['role' => UserType::USER]);
+        $this->actingAs($user1);
+        $user = factory(User::class)->create(
+            [
+                'name' => "Name Create",
```
Comment (framgiaciv3) [inline app/Http/Controllers/UserController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Strings.DoubleQuoteUsage.NotRequired</br>**Severity**: error</br>**File**: app/Http/Controllers/UserController.php</br>**Line**: 185</br>String "Name Create" does not require double quotes; use single quotes instead

File: app/Http/Controllers/UserController.php [hunk 6]
```diff
@@ -125,4 +125,339 @@ public function destroy(User $user)
...
+                'email' => 'email@gmail.com',
+                'password' => '12345678',
+                'role' => '1',
+                'avatar' => UploadedFile::fake()->image('avatar.jpg'),
+            ]
+        );
+        $params = [
+            'name' => "Name Update",
```
Comment (framgiaciv3) [inline app/Http/Controllers/UserController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Strings.DoubleQuoteUsage.NotRequired</br>**Severity**: error</br>**File**: app/Http/Controllers/UserController.php</br>**Line**: 193</br>String "Name Update" does not require double quotes; use single quotes instead

File: app/Http/Controllers/UserController.php [hunk 7]
```diff
@@ -125,4 +125,339 @@ public function destroy(User $user)
...
+        $response->assertStatus(302);
+    }
+
+    public function testUpdateUserRequireName()
+    {
+        $admin = factory(User::class)->create(['role' => UserType::ADMIN]);
+        $this->actingAs($admin);
+        $user = factory(User::class)->create(
+            [
+                'name' => "Name Create",
```
Comment (framgiaciv3) [inline app/Http/Controllers/UserController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Strings.DoubleQuoteUsage.NotRequired</br>**Severity**: error</br>**File**: app/Http/Controllers/UserController.php</br>**Line**: 209</br>String "Name Create" does not require double quotes; use single quotes instead

File: app/Http/Controllers/UserController.php [hunk 8]
```diff
@@ -125,4 +125,339 @@ public function destroy(User $user)
...
+            ->assertSessionHasErrors('name');
+    }
+
+    public function testUpdateUserNameMinLength()
+    {
+        $admin = factory(User::class)->create(['role' => UserType::ADMIN]);
+        $this->actingAs($admin);
+        $user = factory(User::class)->create(
+            [
+                'name' => "Name Create",
```
Comment (framgiaciv3) [inline app/Http/Controllers/UserController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Strings.DoubleQuoteUsage.NotRequired</br>**Severity**: error</br>**File**: app/Http/Controllers/UserController.php</br>**Line**: 235</br>String "Name Create" does not require double quotes; use single quotes instead

File: app/Http/Controllers/UserController.php [hunk 9]
```diff
@@ -125,4 +125,339 @@ public function destroy(User $user)
...
+            ->assertSessionHasErrors('name');
+    }
+
+    public function testUpdateUserNameMaximumLength()
+    {
+        $admin = factory(User::class)->create(['role' => UserType::ADMIN]);
+        $this->actingAs($admin);
+        $user = factory(User::class)->create(
+            [
+                'name' => "Name Create",
```
Comment (framgiaciv3) [inline app/Http/Controllers/UserController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Strings.DoubleQuoteUsage.NotRequired</br>**Severity**: error</br>**File**: app/Http/Controllers/UserController.php</br>**Line**: 261</br>String "Name Create" does not require double quotes; use single quotes instead

File: app/Http/Controllers/UserController.php [hunk 10]
```diff
@@ -125,4 +125,339 @@ public function destroy(User $user)
...
+            ->assertSessionHasErrors('name');
+    }
+
+    public function testUpdateUserRequireEmail()
+    {
+        $admin = factory(User::class)->create(['role' => UserType::ADMIN]);
+        $this->actingAs($admin);
+        $user = factory(User::class)->create(
+            [
+                'name' => "Name Create",
```
Comment (framgiaciv3) [inline app/Http/Controllers/UserController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Strings.DoubleQuoteUsage.NotRequired</br>**Severity**: error</br>**File**: app/Http/Controllers/UserController.php</br>**Line**: 287</br>String "Name Create" does not require double quotes; use single quotes instead

File: app/Http/Controllers/UserController.php [hunk 11]
```diff
@@ -125,4 +125,339 @@ public function destroy(User $user)
...
+        $response
+            ->assertStatus(302)
+            ->assertSessionHasErrors('email');
+    }
+
+    public function testUpdateUserUniqueEmail()
+    {
+        $admin = factory(User::class)->create(['role' => UserType::ADMIN]);
+        $this->actingAs($admin);
+        $user1 = factory(User::class)->create(['email' => "emailduplicate@gmail.com"]);
```
Comment (framgiaciv3) [inline app/Http/Controllers/UserController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Strings.DoubleQuoteUsage.NotRequired</br>**Severity**: error</br>**File**: app/Http/Controllers/UserController.php</br>**Line**: 311</br>String "emailduplicate@gmail.com" does not require double quotes; use single quotes instead

File: app/Http/Controllers/UserController.php [hunk 12]
```diff
@@ -125,4 +125,339 @@ public function destroy(User $user)
...
+        $user = factory(User::class)->create(
+            [
+                'name' => "Name Create",
```
Comment (framgiaciv3) [inline app/Http/Controllers/UserController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Strings.DoubleQuoteUsage.NotRequired</br>**Severity**: error</br>**File**: app/Http/Controllers/UserController.php</br>**Line**: 314</br>String "Name Create" does not require double quotes; use single quotes instead

File: app/Http/Controllers/UserController.php [hunk 13]
```diff
@@ -125,4 +125,339 @@ public function destroy(User $user)
...
+            ->assertSessionHasErrors('email');
+    }
+
+    public function testUpdateUserEmailMaximumLength()
+    {
+        $admin = factory(User::class)->create(['role' => UserType::ADMIN]);
+        $this->actingAs($admin);
+        $user = factory(User::class)->create(
+            [
+                'name' => "Name Create",
```
Comment (framgiaciv3) [inline app/Http/Controllers/UserController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Strings.DoubleQuoteUsage.NotRequired</br>**Severity**: error</br>**File**: app/Http/Controllers/UserController.php</br>**Line**: 340</br>String "Name Create" does not require double quotes; use single quotes instead

File: app/Http/Controllers/UserController.php [hunk 14]
```diff
@@ -125,4 +125,339 @@ public function destroy(User $user)
...
+            ->assertSessionHasErrors('email');
+    }
+
+    public function testUpdateUserRequirePassword()
+    {
+        $admin = factory(User::class)->create(['role' => UserType::ADMIN]);
+        $this->actingAs($admin);
+        $user = factory(User::class)->create(
+            [
+                'name' => "Name Create",
```
Comment (framgiaciv3) [inline app/Http/Controllers/UserController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Strings.DoubleQuoteUsage.NotRequired</br>**Severity**: error</br>**File**: app/Http/Controllers/UserController.php</br>**Line**: 366</br>String "Name Create" does not require double quotes; use single quotes instead

File: app/Http/Controllers/UserController.php [hunk 15]
```diff
@@ -125,4 +125,339 @@ public function destroy(User $user)
...
+            ->assertSessionHasErrors('password');
+    }
+
+    public function testUpdateUserRequirePasswordMinLength()
+    {
+        $admin = factory(User::class)->create(['role' => UserType::ADMIN]);
+        $this->actingAs($admin);
+        $user = factory(User::class)->create(
+            [
+                'name' => "Name Create",
```
Comment (framgiaciv3) [inline app/Http/Controllers/UserController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Strings.DoubleQuoteUsage.NotRequired</br>**Severity**: error</br>**File**: app/Http/Controllers/UserController.php</br>**Line**: 392</br>String "Name Create" does not require double quotes; use single quotes instead

File: app/Http/Controllers/UserController.php [hunk 16]
```diff
@@ -125,4 +125,339 @@ public function destroy(User $user)
...
+            ->assertSessionHasErrors('password');
+    }
+
+    public function testUpdateUserRequirePasswordMaximumLength()
+    {
+        $admin = factory(User::class)->create(['role' => UserType::ADMIN]);
+        $this->actingAs($admin);
+        $user = factory(User::class)->create(
+            [
+                'name' => "Name Create",
```
Comment (framgiaciv3) [inline app/Http/Controllers/UserController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Strings.DoubleQuoteUsage.NotRequired</br>**Severity**: error</br>**File**: app/Http/Controllers/UserController.php</br>**Line**: 418</br>String "Name Create" does not require double quotes; use single quotes instead

File: app/Http/Controllers/UserController.php [hunk 17]
```diff
@@ -125,4 +125,339 @@ public function destroy(User $user)
...
+            ->assertSessionHasErrors('password');
+    }
+    
+    public function testUpdateUserAvatarNotImage()
+    {
+        $admin = factory(User::class)->create(['role' => UserType::ADMIN]);
+        $this->actingAs($admin);
+        $user = factory(User::class)->create(
+            [
+                'name' => "Name Create",
```
Comment (framgiaciv3) [inline app/Http/Controllers/UserController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Strings.DoubleQuoteUsage.NotRequired</br>**Severity**: error</br>**File**: app/Http/Controllers/UserController.php</br>**Line**: 444</br>String "Name Create" does not require double quotes; use single quotes instead

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr></table>

**Review by tamtt-1031 — APPROVED** [review]

**Comment by phucqd-1305** [comment]
OK

**Pull request merged**